src/components/chat/Chat.js

```diff
@@ -15,37 +15,57 @@ class Chat extends React.Component {
         this.state = {
             messages: [],
             loaded: false,
-            loadMessagesTimer: null
+            loadMessagesTimer: null,
+            asyncLock: false
         }
         this.loadChatMessages = this.loadChatMessages.bind(this);
     }
 
 
     async componentDidMount() {
-        await this.loadChatMessages();
-
-        let timer = setInterval(this.loadChatMessages, 1000);
-        this.setState({loadMessagesTimer: timer});
-
+         try {
+            let requestHeader = 'X-Auth-Token ' + localStorage.getItem('token');
+            let response = await api.get(`chat/${this.props.chatEndpoint}`, {headers: {'X-Auth-Token': requestHeader}});
```

Comment:
The chat endpoint is not working on my laptop. Are you sure they work?
They go to the endpoint '/chat' and not '/chat/{id}'

Reply:
And also when sending a message the put request is rejected

Reply:
I did not change to the sessionStorage before, fixed that now.